google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java

```diff
@@ -45,6 +53,17 @@
     T run(DatastoreReaderWriter readerWriter) throws Exception;
   }
 
+  /**
+   * A wrapper around {@link TransactionCallable} with {@link TransactionOptions}.
+   * If {@link TransactionOptions} mode is set to ReadWrite, its previous transaction id will
+   * be populated each time the underlying {@link TransactionCallable} is retried.
+   *
+   * @param <T> the type of the return value
+   */
+  interface ReadWriteTransactionCallable<T> {
```

Comment:
It does not look like this interface should be exposed. For example, both versions of runIntransaction() method (with and without options) accept `TransactionCallable<>`. It like the wrapper is an implementation detail (and should be). If that is the case, please remove it from the client's surface.

Reply:
I have removed `ReadWriteTransactionCallable` in the interface.